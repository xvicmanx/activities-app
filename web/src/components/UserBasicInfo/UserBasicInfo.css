.UserBasicInfo {
  padding: 20px;
  display: flex;
  align-items: center;
  justify-content: center;
}

.UserBasicInfo__Picture {
  width: 25%;
  margin-right: 20px;
}

.UserBasicInfo__Picture img {
  border-radius: 50%;
}

.UserBasicInfo .inverted {
  color: #909aa1;
  padding: 10px 0 0 0;
  text-align: center;
}


@media only screen and (min-width: 1024px) {
  .UserBasicInfo {
    flex-direction: column;
  }

  .UserBasicInfo__Picture {
    display: block;
    width: 50%;
    margin-right: 0;
  }

  .UserBasicInfo__Picture img {
    border-radius: 50%;
    margin: auto;
    display: block;
  }
}