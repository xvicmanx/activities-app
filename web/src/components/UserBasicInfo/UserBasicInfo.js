import React from 'react';
import { Heading, Image } from 'react-bulma-components';

import './UserBasicInfo.css';

const UserBasicInfo = ({ image, user }) => {
  if (!user) return null;
  return (
    <div className="UserBasicInfo">
      <Image
        src={image}
        alt="Pic"
        className="UserBasicInfo__Picture"
      />
      <div>
        <Heading
          size={4}
          className="inverted"
        >
          {user.name}
        </Heading>
      </div>
    </div>
  );
};

export default UserBasicInfo;
