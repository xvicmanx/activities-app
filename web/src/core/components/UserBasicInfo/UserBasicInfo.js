// @flow

import React from 'react';
import { Heading, Image } from 'react-bulma-components';

import './UserBasicInfo.css';

type User = {
  profileURL: string,
  name: string,
};
type Props = {
  user?: User,
};

const UserBasicInfo = ({ user }: Props): null | React$Element<'div'> => {
  if (!user) return null;
  return (
    <div className="UserBasicInfo">
      <Image
        src={user.profileURL}
        alt="Pic"
        className="UserBasicInfo__Picture"
      />
      <div>
        <Heading size={4} className="inverted">
          {user.name}
        </Heading>
      </div>
    </div>
  );
};

export default UserBasicInfo;
