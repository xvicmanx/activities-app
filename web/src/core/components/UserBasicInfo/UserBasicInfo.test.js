import React from 'react';
import renderer from 'react-test-renderer';
import UserBasicInfo from './UserBasicInfo';

describe('UserBasicInfo', () => {
  it('renders properly when user is not provided', () => {
    const component = renderer.create(<UserBasicInfo />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders properly when user is provided', () => {
    const user = {
      name: 'John',
      profileURL: 'test.com',
    };
    const component = renderer.create(<UserBasicInfo user={user} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
