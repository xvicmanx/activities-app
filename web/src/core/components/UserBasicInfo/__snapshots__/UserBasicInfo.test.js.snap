// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserBasicInfo renders properly when user is not provided 1`] = `null`;

exports[`UserBasicInfo renders properly when user is provided 1`] = `
<div
  className="UserBasicInfo"
>
  <figure
    className="image UserBasicInfo__Picture"
  >
    <img
      alt="Pic"
      className=""
      onError={[Function]}
      src="test.com"
    />
  </figure>
  <div>
    <h1
      className="inverted title is-4"
    >
      John
    </h1>
  </div>
</div>
`;
